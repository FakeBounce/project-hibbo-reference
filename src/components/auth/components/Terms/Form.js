import React from 'react';
import { View, Dimensions } from 'react-native';
import { shape, string, arrayOf, bool, func } from 'prop-types';

import { contriesList } from 'utils/countries';

import FormInput from 'shared/FormInput';
import DropDownSelector from 'shared/DropdownSelector';

const margin = 40;
const inputWidth = Dimensions.get('window').width - 2 * margin;

const Form = ({
  innerRefs,
  homeAddress,
  birthday,
  ssn,
  labels,
  error,
  onChange,
  onSubmit,
}) => {
  const refs = innerRefs;

  return (
    <View>
      {Object.keys(homeAddress).map((field, index) => {
        if (index === 0) {
          return (
            <DropDownSelector
              key={`${field}${index}`}
              name={labels[index][1]}
              label={labels[index][0]}
              placeholder={labels[index][0]}
              value={homeAddress[labels[index][1]]}
              onChange={onChange}
              list={contriesList}
              error={error}
            />
          );
        } else if (index === 4 && homeAddress.country === 'US') {
          return (
            <FormInput
              key={`${field}${index}`}
              value={homeAddress[labels[index + 1][1]]}
              label={labels[index + 1][0]}
              name={labels[index + 1][1]}
              onChange={onChange}
              onSubmit={() => {
                if (homeAddress[field] !== '') {
                  refs[labels[index + 2][1]].focus();
                }
                onSubmit(labels[index][1]);
              }}
              placeholder={labels[index + 1][0]}
              keyboardType={index === 2 ? 'numeric' : 'default'}
              innerRef={element => {
                refs[labels[index][1]] = element;
              }}
              error={error}
              inputWidth={inputWidth}
            />
          );
        }

        return (
          <FormInput
            key={`${field}${index}`}
            value={homeAddress[labels[index][1]]}
            label={labels[index][0]}
            name={labels[index][1]}
            onChange={onChange}
            onSubmit={() => {
              if (homeAddress[field] !== '') {
                refs[labels[index + 1][1]].focus();
              }
              onSubmit(labels[index][1]);
            }}
            placeholder={labels[index][0]}
            keyboardType={index === 2 ? 'numeric' : 'default'}
            innerRef={element => {
              refs[labels[index][1]] = element;
            }}
            error={error}
            inputWidth={inputWidth}
          />
        );
      })}
      <FormInput
        value={birthday}
        label={labels[6][0]}
        name={labels[6][1]}
        onChange={onChange}
        onSubmit={onSubmit}
        placeholder={labels[6][0]}
        keyboardType="numeric"
        innerRef={element => {
          refs.birthday = element;
        }}
        error={error}
        maxLength={10}
        inputWidth={inputWidth}
      />
      {homeAddress.country.toUpperCase() === 'US' && (
        <FormInput
          value={ssn}
          label={labels[7][0]}
          name={labels[7][1]}
          onChange={onChange}
          onSubmit={onSubmit}
          placeholder={labels[7][0]}
          returnKeyType="done"
          keyboardType="numeric"
          innerRef={element => {
            refs.ssn = element;
          }}
          error={error}
          maxLength={4}
          inputWidth={inputWidth}
        />
      )}
    </View>
  );
};

Form.propTypes = {
  homeAddress: shape({
    country: string.isRequired,
    address: string.isRequired,
    postalCode: string.isRequired,
    city: string.isRequired,
    region: string.isRequired,
  }).isRequired,
  birthday: string.isRequired,
  ssn: string.isRequired,
  labels: arrayOf(arrayOf(string)).isRequired,
  error: bool.isRequired,
  onChange: func.isRequired,
  onSubmit: func.isRequired,
  innerRefs: shape({}).isRequired,
};

export default Form;
