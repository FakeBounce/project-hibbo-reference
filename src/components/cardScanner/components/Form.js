import React from 'react';
import { Dimensions, StyleSheet } from 'react-native';
import { shape, string, func, arrayOf, bool } from 'prop-types';

import FormInput from 'shared/FormInput';
import Input from 'shared/Input';
import DropdownSelector from 'shared/DropdownSelector';

import { StyledContainerRow } from 'styledComponents/containers';
import { countriesCurrencies } from 'utils/countries';

import { SForm } from '../styles';

const { width } = Dimensions.get('window');
const styles = StyleSheet.create({
  date: {
    marginRight: 15,
  },
  year: {
    marginTop: -4,
  },
});

let cardRef = null;
let monthRef = null;
let yearRef = null;
let cryptoRef = null;

const Form = ({ form, label, onChange, error }) => {
  return (
    <SForm>
      <FormInput
        label={label[0]}
        placeholder={label[0]}
        name="name"
        onChange={onChange}
        onSubmit={() => {
          cardRef.focus();
        }}
        error={error.name}
        inputWidth={width - 80}
        value={form.name}
      />

      <DropdownSelector
        name="currency"
        label={label[5]}
        value={form.currency}
        placeholder={label[5]}
        list={countriesCurrencies}
        onChange={onChange}
        error={error.currency}
      />

      <FormInput
        innerRef={element => {
          cardRef = element;
        }}
        label={label[1]}
        placeholder={label[1]}
        name="cardNumber"
        onChange={(name, text) => {
          if (text.length > 15) {
            monthRef.focus();
          }
          return onChange(name, text);
        }}
        error={error.cardNumber}
        inputWidth={width - 80}
        value={form.cardNumber}
        maxLength={16}
        keyboardType="numeric"
      />

      <StyledContainerRow>
        <FormInput
          innerRef={element => {
            monthRef = element;
          }}
          label={label[2]}
          placeholder={label[2]}
          name="month"
          onChange={(name, text) => {
            if (text.length > 1) {
              yearRef.focus();
            }
            return onChange(name, text);
          }}
          error={error.month}
          inputWidth={50}
          style={styles.date}
          value={form.month}
          maxLength={2}
          keyboardType="numeric"
        />
        <Input
          innerRef={element => {
            yearRef = element;
          }}
          label={label[3]}
          placeholder={label[3]}
          name="year"
          onChange={(name, text) => {
            if (text.length > 1) {
              cryptoRef.focus();
            }
            return onChange(name, text);
          }}
          error={error.year}
          inputWidth={50}
          value={form.year}
          maxLength={4}
          keyboardType="numeric"
          style={styles.year}
        />
        <FormInput
          innerRef={element => {
            cryptoRef = element;
          }}
          label={label[4]}
          placeholder={label[4]}
          name="crypto"
          onChange={onChange}
          error={error.crypto}
          inputWidth={70}
          value={form.crypto}
          maxLength={3}
          keyboardType="numeric"
        />
      </StyledContainerRow>
    </SForm>
  );
};

Form.propTypes = {
  form: shape({
    name: string.isRequired,
    cardNumber: string.isRequired,
    cardType: string.isRequired,
    month: string.isRequired,
    year: string.isRequired,
  }).isRequired,
  onChange: func.isRequired,
  label: arrayOf(string).isRequired,
  error: shape({
    name: bool.isRequired,
    cardNumber: bool.isRequired,
    cardType: bool.isRequired,
    month: bool.isRequired,
    year: bool.isRequired,
  }).isRequired,
};

export default Form;
