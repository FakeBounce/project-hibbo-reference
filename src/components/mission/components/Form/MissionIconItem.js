import React from 'react';
import { StyleSheet, Image } from 'react-native';
import { shape, string, bool } from 'prop-types';
import appStyles from 'styles/appStyles';
import IconSelect from 'assets/mission/IconSelect.png';
import IconSelectError from 'assets/mission/iconSelectError.png';
import { StyledModalIconPanelItemSelectIcon } from '../../styles';

const styles = StyleSheet.create({
  imageStyle: {
    maxWidth: 120,
    height: 48,
  },
});

const colorTheme = {};
appStyles.colorList.map((color, index) => {
  colorTheme[color] = appStyles.colorListSuffix[index];
  return true;
});

const MissionIconItem = ({ icon, theme, error }) => {
  let iconItem = error ? (
    <Image source={IconSelectError} />
  ) : (
    <Image source={IconSelect} />
  );

  let iconName = icon.icon;
  if (typeof colorTheme[theme] !== 'undefined') iconName += colorTheme[theme];
  const iconObject = { icon: iconName };

  if (icon.icon !== '') {
    iconItem = (
      <StyledModalIconPanelItemSelectIcon
        icon={iconObject}
        imageStyle={styles.imageStyle}
      />
    );
  }

  return iconItem;
};

MissionIconItem.propTypes = {
  icon: shape({
    icon: string.isRequired,
    buyable: shape({
      price: string,
      color: string,
      name: string,
    }),
  }).isRequired,
  error: bool.isRequired,
  theme: string.isRequired,
};

export default MissionIconItem;
