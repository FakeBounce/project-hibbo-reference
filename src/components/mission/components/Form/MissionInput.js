import React, { PureComponent } from 'react';
import { bind } from 'decko';
import { Animated } from 'react-native';
import {
  string,
  func,
  bool,
  number,
  oneOfType,
  shape,
  instanceOf,
} from 'prop-types';
import appStyles from 'styles/appStyles';
import { SAnimatedWhiteBorderedinput, STextInput } from '../../styles';

class MissionInput extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isFocused: false,
      placeholderText: this.props.label,
      opacity: new Animated.Value(0.3),
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.value.length !== this.props.value.length) {
      if (nextProps.value.length === 0) this.resetLabel(nextProps);
      else
        this.setState(state => ({
          ...state,
          opacity: new Animated.Value(1),
        }));
    }
  }

  @bind
  onToggle() {
    this.setState(
      state => ({
        ...state,
        isFocused: !state.isFocused,
      }),
      () => {
        Animated.timing(this.state.opacity, {
          toValue:
            this.props.value.toString().length > 0 // eslint-disable-line
              ? 1
              : this.state.isFocused ? 0 : 0.3,
        }).start(() => {
          this.resetLabel(this.props);
        });
      },
    );
  }

  @bind
  resetLabel(props) {
    this.setState(
      state => ({
        ...state,
        placeholderText: state.isFocused ? '' : props.label,
      }),
      () => {
        Animated.timing(this.state.opacity, {
          toValue: 1,
        }).start();
      },
    );
  }

  render() {
    const {
      width,
      animation,
      error,
      value,
      onChange,
      maxLength,
      focus,
      keyboard,
      innerRef,
      name,
      onSubmitEditing,
    } = this.props;
    const { isFocused, placeholderText, opacity } = this.state;
    let val = value;
    if (keyboard !== 'numeric') val = value !== 0 ? value.toString() : '';
    return (
      <SAnimatedWhiteBorderedinput
        focused={isFocused}
        sWidth={width}
        style={{ opacity: animation.opacityBorder }}
        error={error}
      >
        <Animated.View
          style={[
            {
              opacity: animation.opacity,
              transform: [
                { translateY: animation.translateY },
                { scaleY: animation.scaleY },
              ],
            },
            { opacity },
          ]}
        >
          <STextInput
            keyboardType={keyboard}
            returnKeyType="next"
            value={val}
            autoCapitalize="sentences"
            maxLength={maxLength}
            autoCorrect={false}
            autoFocus={focus}
            onChangeText={text => {
              if (!isFocused) {
                this.onToggle();
              }
              onChange(name, text);
            }}
            onFocus={this.onToggle}
            onBlur={this.onToggle}
            placeholder={placeholderText}
            placeholderTextColor={
              error ? appStyles.colors.error : appStyles.colors.placeholderGrey
            }
            sWidth={width}
            color={error ? appStyles.colors.error : appStyles.colors.white}
            underlineColorAndroid="transparent"
            innerRef={innerRef}
            onSubmitEditing={onSubmitEditing}
          />
        </Animated.View>
      </SAnimatedWhiteBorderedinput>
    );
  }
}

MissionInput.defaultProps = {
  keyboard: 'default',
  focus: false,
  maxLength: 50,
  innerRef: null,
  onSubmitEditing: null,
};

MissionInput.propTypes = {
  innerRef: func,
  name: string.isRequired,
  label: string.isRequired,
  onChange: func.isRequired,
  value: oneOfType([string, number]).isRequired,
  error: bool.isRequired,
  width: number.isRequired,
  maxLength: number,
  keyboard: string,
  focus: bool,
  onSubmitEditing: func,
  animation: shape({
    translateY: instanceOf(Animated.Value).isRequired,
    opacity: instanceOf(Animated.Value).isRequired,
    opacityBorder: instanceOf(Animated.Value).isRequired,
    scaleY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
};

export default MissionInput;
