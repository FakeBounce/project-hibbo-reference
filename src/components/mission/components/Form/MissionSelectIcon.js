import React from 'react';
import { Animated } from 'react-native';
import { shape, func, string, bool, instanceOf } from 'prop-types';
import Select from 'assets/mission/select.png';
import SelectError from 'assets/mission/selectError.png';
import MissionIconItem from './MissionIconItem';
import {
  StyledTouchableOpacityIcon,
  StyledImageSelectIcon,
  SAnimatedContainerDeepBlue,
} from '../../styles';

const MissionSelectIcon = ({ onPress, icon, theme, error, animation }) => {
  let width = 80;
  if (icon.icon !== '') width = 155;
  return (
    <SAnimatedContainerDeepBlue
      sWidth={width}
      style={{
        backgroundColor: theme, // Does not render through SC
        opacity: animation.opacity,
        transform: [
          { translateY: animation.translateY },
          { scaleY: animation.scaleY },
        ],
      }}
    >
      <StyledTouchableOpacityIcon onPress={onPress} sWidth={width}>
        <MissionIconItem icon={icon} theme={theme} error={error} />
        <StyledImageSelectIcon source={error ? SelectError : Select} />
      </StyledTouchableOpacityIcon>
    </SAnimatedContainerDeepBlue>
  );
};

MissionSelectIcon.propTypes = {
  onPress: func.isRequired,
  theme: string.isRequired,
  icon: shape({
    icon: string.isRequired,
    buyable: shape({
      price: string,
      color: string,
      name: string,
    }),
  }).isRequired,
  error: bool.isRequired,
  animation: shape({
    translateY: instanceOf(Animated.Value).isRequired,
    opacity: instanceOf(Animated.Value).isRequired,
    scaleY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
};

export default MissionSelectIcon;
