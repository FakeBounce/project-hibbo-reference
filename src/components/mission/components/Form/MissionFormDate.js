import React from 'react';
import { Animated } from 'react-native';
import { string, bool, func, shape, instanceOf } from 'prop-types';
import { StyledFlexRow } from '../../styles';

import MissionInput from './MissionInput';

let secondInput = null;
let thirdInput = null;
const FormDate = ({
  day,
  month,
  year,
  onChange,
  onSubmitEditing,
  ...otherProps
}) => {
  return (
    <StyledFlexRow>
      <MissionInput
        {...day}
        {...otherProps}
        width={41}
        keyboard="numeric"
        maxLength={2}
        onChange={(name, text) => {
          if (Number(text) <= 31 || text === '') {
            onChange(name, text);
            if (text.length === 2 && Number(text) > 0 && secondInput)
              secondInput.focus();
          }
        }}
      />

      <MissionInput
        {...month}
        {...otherProps}
        width={55}
        keyboard="numeric"
        maxLength={2}
        innerRef={element => {
          secondInput = element;
        }}
        onChange={(name, text) => {
          if (Number(text) <= 12 || text === '') {
            onChange(name, text);
            if (text.length === 2 && Number(text) > 0 && thirdInput)
              thirdInput.focus();
          }
        }}
      />

      <MissionInput
        {...year}
        {...otherProps}
        width={75}
        keyboard="numeric"
        maxLength={4}
        innerRef={element => {
          thirdInput = element;
        }}
        onChange={(name, text) => {
          const t = new Date();
          if (text.length === 4 && Number(text) >= t.getFullYear())
            onSubmitEditing();
          onChange(name, text);
        }}
      />
    </StyledFlexRow>
  );
};

FormDate.defaultProps = {
  onSubmitEditing: null,
};

FormDate.propTypes = {
  day: shape({
    label: string.isRequired,
    name: string.isRequired,
    value: string.isRequired,
    error: bool.isRequired,
    animation: shape({
      translateY: instanceOf(Animated.Value),
      opacity: instanceOf(Animated.Value),
      opacityBorder: instanceOf(Animated.Value),
      scaleY: instanceOf(Animated.Value),
    }).isRequired,
  }).isRequired,
  month: shape({
    label: string.isRequired,
    name: string.isRequired,
    value: string.isRequired,
    error: bool.isRequired,
    animation: shape({
      translateY: instanceOf(Animated.Value),
      opacity: instanceOf(Animated.Value),
      opacityBorder: instanceOf(Animated.Value),
      scaleY: instanceOf(Animated.Value),
    }).isRequired,
  }).isRequired,
  year: shape({
    label: string.isRequired,
    name: string.isRequired,
    value: string.isRequired,
    error: bool.isRequired,
    animation: shape({
      translateY: instanceOf(Animated.Value),
      opacity: instanceOf(Animated.Value),
      opacityBorder: instanceOf(Animated.Value),
      scaleY: instanceOf(Animated.Value),
    }).isRequired,
  }).isRequired,
  onChange: func.isRequired,
  onSubmitEditing: func,
};

export default FormDate;
