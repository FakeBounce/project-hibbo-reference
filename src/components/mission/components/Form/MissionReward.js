import React from 'react';
import { Animated } from 'react-native';
import { string, bool, func, instanceOf, shape } from 'prop-types';
import { StyledContainerRow } from 'styles/styledComponents/containers';
import { SAnimatedWhiteEuro } from '../../styles';
import MissionInput from './MissionInput';

const MissionReward = ({
  form,
  label,
  name,
  error,
  currency,
  onChange,
  innerRef,
  animation,
}) => {
  const width = 35 + form.length * 10;
  return (
    <StyledContainerRow>
      <MissionInput
        value={form}
        label={label}
        name={name}
        onChange={(changeName, text) => {
          let txt = text;
          if (text.length === 4) {
            if (form.length === 5) {
              txt = text.substring(0, 3);
            } else {
              txt =`${text},`;
            }
          }
          onChange(changeName, txt);
        }}
        error={error}
        width={width}
        maxLength={7}
        innerRef={innerRef}
        keyboard="numeric"
        animation={animation}
      />
      <SAnimatedWhiteEuro
        error={error}
        style={{
          opacity: animation.opacity,
          transform: [
            { translateY: animation.translateY },
            { scaleY: animation.scaleY },
          ],
        }}
      >
        {currency}
      </SAnimatedWhiteEuro>
    </StyledContainerRow>
  );
};

MissionReward.defaultProps = {
  innerRef: null,
  currency: 'â‚¬',
};

MissionReward.propTypes = {
  form: string.isRequired,
  error: bool.isRequired,
  label: string.isRequired,
  name: string.isRequired,
  onChange: func.isRequired,
  innerRef: func,
  animation: shape({
    translateY: instanceOf(Animated.Value).isRequired,
    opacity: instanceOf(Animated.Value).isRequired,
    scaleY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  currency: string,
};

export default MissionReward;
