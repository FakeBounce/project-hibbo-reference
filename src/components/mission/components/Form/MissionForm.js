import React, { PureComponent } from 'react';
import { bind } from 'decko';
import { Keyboard, StyleSheet, Animated, Easing } from 'react-native';
import { string, shape, func, bool, arrayOf } from 'prop-types';
import { getTranslations } from 'utils/i18n';
import { GreyRoundedButton } from 'shared/Button';
import FormDate from './MissionFormDate';
import MissionInput from './MissionInput';
import MissionReward from './MissionReward';
import MissionSelectIcon from './MissionSelectIcon';
import {
  StyledMargedForm,
  StyledSpacedRow,
  SAnimatedCenteredMargedTopSpacedRow,
} from '../../styles';

const styles = StyleSheet.create({
  styleButton: { height: 56, width: 104, marginTop: 70 },
});

let nextInput = null;
let nextInputAmount = null;
class MissionForm extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isFrontFace: false,
    };

    this.animations = {
      dayAnimation: {},
      monthAnimation: {},
      yearAnimation: {},
      titleAnimation: {},
      rewardAnimation: {},
      iconAnimation: {},
      buttonAnimation: {},
    };

    this.resetAnimation();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.isFrontFace !== this.state.isFrontFace) {
      this.setState(
        state => ({
          ...state,
          isFrontFace: nextProps.isFrontFace,
        }),
        () => {
          if (nextProps.isFrontFace) {
            Animated.delay(500).start(() => {
              Animated.stagger(70, [
                this.staggeredAnimation(this.animations.dayAnimation),
                this.staggeredAnimation(this.animations.monthAnimation),
                this.staggeredAnimation(this.animations.yearAnimation),
                this.staggeredAnimation(this.animations.titleAnimation),
                this.staggeredAnimation(this.animations.rewardAnimation),
                this.staggeredAnimation(this.animations.iconAnimation),
              ]).start();

              Animated.delay(500).start(() => {
                this.staggeredAnimation(
                  this.animations.buttonAnimation,
                ).start();
              });
            });
          } else {
            this.resetAnimation();
          }
        },
      );
    }
  }

  @bind
  staggeredAnimation({ opacityBorder, opacity, translateY, scaleY }) {
    return Animated.stagger(100, [
      Animated.timing(opacityBorder, {
        toValue: 1,
        duration: 120,
        easing: Easing.in,
      }),
      Animated.parallel([
        Animated.timing(opacity, {
          toValue: 1,
          duration: 90,
          easing: Easing.in,
        }),
        Animated.timing(translateY, {
          toValue: 0,
          duration: 270,
          easing: Easing.in(Easing.bounce),
        }),
        Animated.timing(scaleY, {
          toValue: 1,
          duration: 90,
          easing: Easing.in,
        }),
      ]),
    ]);
  }

  @bind
  resetAnimation() {
    Object.keys(this.animations).forEach(key => {
      this.animations[key] = {
        translateY: new Animated.Value(20),
        opacity: new Animated.Value(0),
        opacityBorder: new Animated.Value(0),
        scaleY: new Animated.Value(0),
      };
    });
  }

  render() {
    const {
      label,
      form,
      error,
      name,
      theme,
      icon,
      editMode,
      onChange,
      onAddMission,
      onPress,
    } = this.props;
    return (
      <StyledMargedForm>
        <StyledSpacedRow>
          <FormDate
            day={{
              label: label[0],
              name: name[0],
              error: error.day,
              value: form.day,
              animation: this.animations.dayAnimation,
            }}
            month={{
              label: label[1],
              name: name[1],
              error: error.month,
              value: form.month,
              animation: this.animations.monthAnimation,
            }}
            year={{
              label: label[2],
              name: name[2],
              error: error.year,
              value: form.year,
              animation: this.animations.yearAnimation,
            }}
            onChange={onChange}
            onSubmitEditing={() => {
              nextInput.focus();
            }}
          />
        </StyledSpacedRow>

        <StyledSpacedRow>
          <MissionInput
            value={form.title}
            label={label[3]}
            name={name[3]}
            onChange={onChange}
            error={error.title}
            width={250}
            maxLength={30}
            animation={this.animations.titleAnimation}
            innerRef={element => {
              nextInput = element;
            }}
            onSubmitEditing={() => {
              nextInputAmount.focus();
            }}
          />
        </StyledSpacedRow>
        <StyledSpacedRow>
          <MissionReward
            form={form.amount}
            label={label[4]}
            name={name[4]}
            error={error.amount}
            onChange={onChange}
            animation={this.animations.rewardAnimation}
            innerRef={element => {
              nextInputAmount = element;
            }}
          />
        </StyledSpacedRow>

        <StyledSpacedRow paddingBottom semiRow>
          <MissionSelectIcon
            onPress={() => {
              Keyboard.dismiss();
              onPress();
            }}
            icon={icon}
            theme={theme}
            error={error.icon}
            animation={this.animations.iconAnimation}
          />
        </StyledSpacedRow>

        <SAnimatedCenteredMargedTopSpacedRow
          style={{
            opacity: this.animations.buttonAnimation.opacity,
            transform: [
              { translateY: this.animations.buttonAnimation.translateY },
              { scaleY: this.animations.buttonAnimation.scaleY },
            ],
          }}
        >
          <GreyRoundedButton
            text={
              editMode
                ? getTranslations('mission.edit.label')
                : getTranslations('mission.add.label')
            }
            action={() => {
              Keyboard.dismiss();
              onAddMission();
            }}
            styleButton={styles.styleButton}
          />
        </SAnimatedCenteredMargedTopSpacedRow>
      </StyledMargedForm>
    );
  }
}

MissionForm.defaultProps = {
  editMode: false,
};

MissionForm.propTypes = {
  form: shape({
    day: string.isRequired,
    month: string.isRequired,
    year: string.isRequired,
    title: string.isRequired,
    amount: string.isRequired,
  }).isRequired,
  editMode: bool,
  onChange: func.isRequired,
  onPress: func.isRequired,
  label: arrayOf(string).isRequired,
  name: arrayOf(string).isRequired,
  icon: shape({
    icon: string.isRequired,
    buyable: shape({
      price: string,
      color: string,
      name: string,
    }),
  }).isRequired,
  error: shape({
    day: bool.isRequired,
    month: bool.isRequired,
    year: bool.isRequired,
    title: bool.isRequired,
    amount: bool.isRequired,
    icon: bool.isRequired,
  }).isRequired,
  onAddMission: func.isRequired,
  theme: string.isRequired,
  isFrontFace: bool.isRequired,
};

export default MissionForm;
