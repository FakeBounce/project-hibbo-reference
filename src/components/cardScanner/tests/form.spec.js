import React from 'react';
import renderer from 'react-test-renderer';

import Form from '../components/Form';

describe('Form', () => {
  const fn = jest.fn();
  const form = {
    name: '',
    cardNumber: '',
    cardType: '',
    month: '',
    year: '',
    crypto: '',
  };
  const label = ['', '', '', '', ''];
  const error = {
    name: false,
    cardNumber: false,
    cardType: false,
    month: false,
    year: false,
    crypto: false,
  };
  const component = renderer.create(
    <Form
      form={form}
      label={label}
      onChange={fn}
      error={error}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
