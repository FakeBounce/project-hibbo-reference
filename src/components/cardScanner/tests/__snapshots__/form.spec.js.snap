// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render the component 1`] = `
<View
  style={undefined}
>
  <View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Card holder
      </Text>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "width": 690,
          },
        ]
      }
    >
      <TextInput
        autoCapitalize="words"
        autoCorrect={false}
        autoFocus={false}
        keyboardType="numeric"
        maxLength={999}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder=""
        placeholderTextColor="#979797"
        returnKeyType="next"
        style={
          Array [
            undefined,
            Object {
              "color": "#413D3D",
              "width": 690,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Card number
      </Text>
    </View>
    <View
      style={
        Array [
          undefined,
          Object {
            "width": 690,
          },
        ]
      }
    >
      <TextInput
        autoCapitalize="words"
        autoCorrect={false}
        autoFocus={false}
        keyboardType="numeric"
        maxLength={16}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder=""
        placeholderTextColor="#979797"
        returnKeyType="next"
        style={
          Array [
            undefined,
            Object {
              "color": "#413D3D",
              "width": 690,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View>
    <View
      style={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Expire at
      </Text>
    </View>
    <View
      style={undefined}
    >
      <View
        style={
          Array [
            undefined,
            Object {
              "width": 50,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="words"
          autoCorrect={false}
          autoFocus={false}
          keyboardType="numeric"
          maxLength={2}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder=""
          placeholderTextColor="#979797"
          returnKeyType="next"
          style={
            Array [
              undefined,
              Object {
                "color": "#413D3D",
                "width": 50,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "width": 50,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="words"
          autoCorrect={false}
          autoFocus={false}
          keyboardType="numeric"
          maxLength={2}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder=""
          placeholderTextColor="#979797"
          returnKeyType="next"
          style={
            Array [
              undefined,
              Object {
                "color": "#413D3D",
                "width": 50,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View>
    <View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={undefined}
        >
          Cryptogram
        </Text>
      </View>
      <View
        style={
          Array [
            undefined,
            Object {
              "width": 70,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="words"
          autoCorrect={false}
          autoFocus={false}
          keyboardType="numeric"
          maxLength={3}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder=""
          placeholderTextColor="#979797"
          returnKeyType="next"
          style={
            Array [
              undefined,
              Object {
                "color": "#413D3D",
                "width": 70,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
</View>
`;
